patches/minecraft/net/minecraft/world/World.java.patch

```diff
@@ -589,23 +610,48 @@
                          if (iblockstate1.func_185904_a() == p_72918_2_)
                          {
                              double d0 = (double)((float)(i4 + 1) - BlockLiquid.func_149801_b(((Integer)iblockstate1.func_177229_b(BlockLiquid.field_176367_b)).intValue()));
-@@ -2116,6 +2287,7 @@
+@@ -2095,8 +2278,23 @@
+             {
+                 for (int j4 = j3; j4 < k3; ++j4)
+                 {
+-                    if (this.func_180495_p(blockpos$pooledmutableblockpos.func_181079_c(l3, i4, j4)).func_185904_a() == p_72875_2_)
++                    
++                    IBlockState iblockstate1 = this.func_180495_p(blockpos$pooledmutableblockpos.func_181079_c(l3, i4, j4));
++                    
++                    Boolean result = iblockstate1.func_177230_c().isAABBInsideLiquid(this, blockpos$pooledmutableblockpos, p_72875_1_);
++                    
++                    if(result != null)
+                     {
++                        if(result)
++                        {
```

Comment:
Please reduce the size of this patch as well, similar to the example I gave for the first patch.